scilpy/io/utils.py

```diff
@@ -155,12 +156,19 @@ def assert_outputs_exist(parser, args, required, optional=None):
     :param required: string or list of paths
     :param optional: string or list of paths.
                      Each element will be ignored if None
+    :param check_dir_exists: test if output directory exists
     """
     def check(path):
         if os.path.isfile(path) and not args.overwrite:
             parser.error('Output file {} exists. Use -f to force '
                          'overwriting'.format(path))
 
+        if check_dir_exists:
+            path_dir = os.path.dirname(path)
+            if path_dir and not os.path.isdir(path_dir):
+                parser.error('Directory {} for a given output file '
```

Comment:
Not sure is needed, but a / could be put after {} since it is a directory.
I actually don't care, it is just in case. Your call

Reply:
done